### PR TITLE
adding push notifications example

### PR DESCRIPTION
**Pull request by tzahie**

### CODE REVIEW

File: Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+                //when getting this action from SampleApplication
+                // that's mean that init of ScheduledNotification finished.
+                if (plusFeature == PlusFeature.SCHEDULED_NOTIFICATIONS) {
+                    scheduledNotificationManager = TaboolaSdkPlus.getScheduledNotificationManager();
+                    if (scheduledNotificationManager != null) {
+                        //init of Scheduled Notification finished successfully
+                        onScheduledNotificationInitFinished();
+                    }
+                } else if (plusFeature == PlusFeature.PUSH_NOTIFICATIONS) {
```
Comment (orenktaboola) [inline Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/MainActivity.java:60]: add comment here too. we need the publisher to undersand the difference between the features

Comment (tzahie) [reply Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/MainActivity.java]: fixed.

File: Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+         /* in order to start displaying notifications,
+         notifications must be enabled via the TBLScheduledManager#enable()
+         and the Taboola News categories must be set
+         via the TBLScheduledManager#setCategories() method */
+
+        /* As a Taboola News publisher, you can choose to allow your user to customize
+         the recommendations displayed on the notification by choosing content categories
+         (News, Sports, Tech, etc.)
+          In order to show a mix of all content categories, set the "general" category. */
+        scheduledNotificationManager.setCategories(Collections.singletonList(CAT_GENERAL));
```
Comment (orenktaboola) [inline Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/MainActivity.java:166]: I think in example code it's better not to write in shortcuts : CATEGORY_GENERAL

File: Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/SampleApplication.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        //init sdk plus
+        initSdkPlus(this);
+    }
+
+
+    private void initSdkPlus(final Context appContext) {
+        SdkPlusPublisherInfo sdkPlusPublisherInfo = new SdkPlusPublisherInfo();
+        sdkPlusPublisherInfo.setConfigId(CONFIG_ID);
+        sdkPlusPublisherInfo.setPublisherName(PUB_NAME);
```
Comment (orenktaboola) [inline Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/SampleApplication.java:42]: PUBLISHER_NAME

File: Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/FBMessagingService.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        boolean isTaboolaMessage = TBLPushManager.isTaboolaMessage(remoteMessage);
+        if (isTaboolaMessage) {
+            // pass the remoteMessage to TBLPushManager so the sdk+ can handle it.
+            TBLPushManager.handleTaboolaPush(remoteMessage);
+        } else {
+            // this is not Taboola push, and you should handle it
+        }
+    }
+
+    @Override
```
Comment (orenktaboola) [inline Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/FBMessagingService.java:42]: i would add some comment here. for example - if there is a new token, we need to pass it to the push manager

File: Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        if (savedInstanceState == null) {
+            // Avoid handling click again when activity is recreated
+            TBLNotificationManager.handleClick(getIntent(), this);
+        }
+
+
+        initLayout();
+
+        scheduledNotificationManager = TaboolaSdkPlus.getScheduledNotificationManager();
+        //if manager is null mean the init still in progress and you should wait for onReceive in the receiver above
```
[the remark this answers is not in the document]
Comment (tzahie) [reply Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/MainActivity.java:88]: fixed

File: Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/MainActivity.java (hunk 2 above)
Comment (tzahie) [reply Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/MainActivity.java:166]: done

File: Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/SampleApplication.java (hunk 3 above)
Comment (tzahie) [reply Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/SampleApplication.java:42]: done

File: Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/FBMessagingService.java (hunk 4 above)
Comment (tzahie) [reply Examples/Push Notificaions Example/app/src/main/java/com/taboola/sample/notification/push/FBMessagingService.java:42]: fixed. 
@orenktaboola please go over it and approve

**Pull request merged**